### PR TITLE
Implement hess_op! and jac_op!

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
This is ready (there is no label).

File: src/NLPModels.jl [hunk 1]
```diff
@@ -160,6 +160,22 @@ function jac_op(nlp :: AbstractNLPModel, x :: Vector{Float64})
                         v -> jtprod(nlp, x, v))
 end
 
+"""`J = jac_op!(nlp, x, Jv, Jtv)`
+
+Return the Jacobian at `x` as a linear operator.
+The resulting object may be used as if it were a matrix, e.g., `J * v` or
+`J' * v`. The values `Jv` and `Jtv` are used to preallocate the memory for the
```
Comment (dpo) [inline src/NLPModels.jl]: Could you just change this to "... are used as preallocated storage for the ..."?

File: src/NLPModels.jl [hunk 2]
```diff
@@ -236,6 +252,25 @@ function hess_op(nlp :: AbstractNLPModel, x :: Vector{Float64};
                         v -> hprod(nlp, x, v; obj_weight=obj_weight, y=y))
 end
 
+"""`H = hess_op!(nlp, x, Hv; obj_weight=1.0, y=zeros)`
+
+Return the Lagrangian Hessian at `(x,y)` with objective function scaled by
+`obj_weight` as a linear operator, and storing the result on `Hv`. The resulting
+object may be used as if it were a matrix, e.g., `w = H * v`. The vector `Hv` is
+used to preallocate the memory for the operation.  The linear operator H
```
Comment (dpo) [inline src/NLPModels.jl]: Same here.

**Comment by dpo** [comment]
Many thanks for this!

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Thank you!

**Pull request merged**